server/src/main/java/org/opensearch/action/search/SearchPhaseController.java

```diff
@@ -615,14 +615,13 @@ private static Sort createSort(TopFieldDocs[] topFieldDocs) {
         final SortField[] newFields = new SortField[firstTopDocFields.length];
 
         for (int i = 0; i < firstTopDocFields.length; i++) {
-            if (firstTopDocFields[i] instanceof SortedNumericSortField && isSortWideningRequired(topFieldDocs, i)) {
-                final SortedNumericSortField delegate = (SortedNumericSortField) firstTopDocFields[i];
-                newFields[i] = new SortedWiderNumericSortField(
-                    delegate.getField(),
-                    delegate.getNumericType(),
-                    delegate.getReverse(),
-                    delegate.getSelector()
-                );
+            final SortField delegate = firstTopDocFields[i];
+            final SortField.Type type = delegate instanceof SortedNumericSortField
+                ? ((SortedNumericSortField) delegate).getNumericType()
+                : delegate.getType();
+
+            if (SortedWiderNumericSortField.isTypeSupported(type) && isSortWideningRequired(topFieldDocs, i)) {
+                newFields[i] = new SortedWiderNumericSortField(delegate.getField(), type, delegate.getReverse());
```

Comment:
Shouldn't we also bring the selector from `SortedNumericSortField` ?

Reply:
Only thing will be used later is ```comparator``` & ```reverse```  from ```SortedWiderNumericSortFieldType``` or ```SortField``` -> [ref](https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/search/TopDocs.java#L162).

I was passing selector earlier in case they modify something inside Lucene and we dont need to change, but now that while serialization of SortedNumericSortField is just SortField, we dont have that info, and seems like we wont need that too in future.